crates/storage/provider/src/bundle_state/bundle_state_with_receipts.rs

```diff
@@ -292,16 +292,14 @@ impl BundleStateWithReceipts {
     /// This plain state will contains some additional information that
     /// are is a artifacts of the lower part state.
     ///
-    /// If block number is in future, return None.
-    pub fn split_at(&mut self, block_number: BlockNumber) -> Option<Self> {
+    /// If block number is not within state range, return [None].
+    pub fn split_after(&mut self, block_number: BlockNumber) -> Option<Self> {
         let last_block = self.last_block();
         let first_block = self.first_block;
-        if block_number >= last_block {
+
+        if block_number < first_block || last_block <= block_number {
```

Comment:
this is fine, because the only time we split the state, we validate that the block is in range
https://github.com/paradigmxyz/reth/blob/deb9e31597ac7e90e68e8329ffcb2d56a744ebc1/crates/storage/provider/src/chain.rs#L203-L231